packages/contracts/test/rbac.ts

```diff
@@ -0,0 +1,460 @@
+
+      let receipt = await client.revokeRole(USER, 1, USER);
+      expect(receipt.success,
+        `Revoking role by ${USER} after granting him role ${role} with "${permission}" permission should be true.`).to.be.true;
+    });
+
+    it("should fail granting permission if called by non-owner user", async () => {
+      const expectedError = "Unauthorized";
+      let receipt = await client.grantPermission("dummy-permission", 1, USER);
+
+      expect(receipt.success, `The success status of granting permission by ${USER} should be false.`).to.be.false;
+      expect(Clarity.unwrapError(receipt),
+      `Granting permission by ${USER} should fail with "${expectedError}" error.`).to.be.eq(expectedError);
+    });
+
+    it("should succeed granting permission if called by contract owner", async () => {
+      let receipt = await client.grantPermission("dummy-permission", 1, DEPLOYER);
+
+      expect(receipt.success, `The success status of granting permission by ${DEPLOYER} should be true.`).to.be.true;
+    });
+
+    it("should succeed granting permission if called by user with correct permission", async () => {
+      const permission = "grant-permission";
+      const role = 10;
+      
+      await client.grantPermission(permission, role, DEPLOYER);
+      await client.grantRole(USER, role, DEPLOYER);
+
+      let receipt = await client.grantPermission("dummy-permission", 1, USER);
+      expect(receipt.success, 
+        `Granting permission by ${USER} after granting him role ${role} with "${permission}" persmission should be true.`).to.be.true;
+    });
+
+    it("should fail revoking permission if called by non-owner user", async () => {
+      const expectedError = "Unauthorized";
+      let receipt = await client.revokePermission("dummy-permission", 1, USER);
+      
+      expect(receipt.success,`The success status of revoking permission by ${USER} should be false.`).to.be.false;
+      expect(Clarity.unwrapError(receipt),
+        `Revoking permission by ${USER} should fail wiht "${expectedError}" error.`).to.be.eq(expectedError);
+    });
+
+    it("should succeed revoking permission if called by contract owner", async () => {
+      await client.grantPermission("dummy-permission", 1, DEPLOYER);
+      
+      let receipt = await client.revokePermission("dummy-permission", 1, DEPLOYER);
+      expect(receipt.success, `The success status of revoking permission by ${DEPLOYER} should be true.`).to.be.true;
+    })
+
+    it("should succeed revoking permission if called by user with correct permission", async () => {
+      const permission = "revoke-permission";
+      const role = 10;
+      
+      await client.grantPermission(permission, role, DEPLOYER);
+      await client.grantRole(USER, role, DEPLOYER);
+      await client.grantPermission("dummy-permission", 1, DEPLOYER);
+      
+      let receipt = await client.revokePermission("dummy-permission", 1, USER);
+      expect(receipt.success, 
```

Comment:
> With clarinet, test could be written in clarity directly.

Can you please show a small code example how it could be done? I've tried `clarinet` and I like it more than official library, but it doesn't work well with Mocha.

> The test should also check for the result, e.g.
> 
> ```
>       expect (receipt.result, "Transaction failed").to.be.eq("Transaction executed and committed. Returned: true\n[]")
> ```

Is it possible that `receipt.success` will be set to `true` and `receipt.result` would tell us something different ?

Reply:
It could for example return `false` instead of `true`

Reply:
Re clarinet: https://github.com/psq/swapr/tree/master/clarinet

Reply:
Those tests are written in JavaScript. With slightly different syntax, full control what you want to include in a specific block, ability to mint empty blocks, read only function calls and what not, but still it's JS,